### PR TITLE
drivers/kw2xrd/Kconfig: fix kconfig model

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

Nightlies are currently failing as there is a difference between make
and kconfig.
This tries to match the kconfig with the makefile dep.
The only issue is the
```
ifneq (,$(filter netdev,$(USEMODULE)))
  USEMODULE += netdev_ieee802154_submac
endif
```

which may have the same effect as select
HAVE_IEEE802154_RADIO_HAL_INTERFACE.

### Testing procedure
`NIGHTLY=1 /bin/bash -c "source .murdock; JOBS=4 compile tests/ieee802154_hal pba-d-01-kw2x:gnu"` should pass.

### Issues/PRs references

Follow-up from #18469

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Like I said, maybe the following should also be in
```diff
diff --git a/boards/pba-d-01-kw2x/Kconfig b/boards/pba-d-01-kw2x/Kconfig
index 8068c000ca..5307eb8a25 100644
--- a/boards/pba-d-01-kw2x/Kconfig
+++ b/boards/pba-d-01-kw2x/Kconfig
@@ -28,3 +28,4 @@ config BOARD_PBA_D_01_KW2X
     select HAVE_MPL3115A2
     select HAVE_TCS37727
     select HAVE_TMP006
+    select HAVE_IEEE802154_RADIO_HAL_INTERFACE

```

If @jia200x could have a quick look and confirm?

**Review by benpicco — APPROVED** [review]

File: drivers/kw2xrf/Kconfig [hunk 1]
```diff
@@ -16,9 +16,6 @@ menuconfig MODULE_KW2XRF
     select MODULE_CORE_THREAD_FLAGS
     select MODULE_IEEE802154
     select MODULE_LUID
-    select MODULE_NETDEV
-    select MODULE_NETDEV_IEEE802154
-    select MODULE_NETDEV_LEGACY_API
```
Comment (benpicco) [inline drivers/kw2xrf/Kconfig:21]: Since those should already be provided by the submac netdev, the removal here is correct either way.

**Comment by MrKevinWeiss** [comment]
I also added this board to the kconfig test in the .murdock so it won't be just nightlies that check it.

**Comment by MrKevinWeiss** [comment]
Just waiting for an opinion on `HAVE_IEEE802154_RADIO_HAL_INTERFACE` since there is a visible difference in the makefile.dep

**Comment by benpicco** [comment]
Why would a board select this? `HAVE_IEEE802154_RADIO_HAL_INTERFACE` is a property of the radio driver.

**Comment by MrKevinWeiss** [comment]
I thought the radio is tied to the board...  @jia200x is in now and I can discuss.

**Comment by benpicco** [comment]
The radio is tied to the board, but `HAVE_IEEE802154_RADIO_HAL_INTERFACE` should be tied to the radio, as it's done with `cc2538`, `nrf802154` and `socket_zep`.

**Comment by MrKevinWeiss** [comment]
Maybe the `MODULE_KW2XRF` selecting it makes more sense.  Ok, pushed the fixup.

**Comment by MrKevinWeiss** [comment]
I guess it looks good.  I squashed and removed the build limiting.

**Comment by MrKevinWeiss** [comment]
Thanks for the review!

**Pull request merged**